fluentd/configs.d/openshift/es-copy-config.conf

```diff
@@ -3,7 +3,7 @@
       host "#{ENV['ES_COPY_HOST']}"
       port "#{ENV['ES_COPY_PORT']}"
       scheme "#{ENV['ES_COPY_SCHEME']}"
-      index_name project.${record['kubernetes']['namespace_name']}.${record['kubernetes']['namespace_id']}.${Time.parse(record['@timestamp']).getutc.strftime(@logstash_dateformat)}
+      index_name project.${begin record['kubernetes']['namespace_name']+'.'+record['kubernetes']['namespace_id']; rescue; $log.error('record is missing kubernetes or kubernetes.namespace_id:' + record.to_s); '.orphan.orphan'; end}.${t = !record['@timestamp'].nil? ? Time.parse(record['@timestamp']) : ($log.error('record is missing @timestamp' + record.to_s);Time.at(time)) ; t.getutc.strftime(@logstash_dateformat)}"
```

Comment:
If we are missing the `@timestamp` field, then the record will not be found at all.  I think we need a filter to ensure every record has a timestamp.

Reply:
We should add that to fluent-plugin-viaq_data_model

Reply:
@portante I'm missing why the record wont be found.  Are you saying it wont be found in the expected index?

Reply:
If a record does not have an @timestamp field, it won't know how to consider that record in the time range it is looking for.

Reply:
We need some way to generate the `yyyy.mm.dd` suffix for the index name for curation, so we need some sort of time field or `@timestamp` to do that.

Reply:
What about if we can not use the `@timstamp` because its missing that we insert it right here?  Does that seem reasonable? What is our alternative?

Reply:
We cannot insert it here - we would need to insert it in a record-transformer filter, or in the viaq filter plugin